x-pack/plugins/observability_solution/synthetics/server/synthetics_service/utils/map_inline_to_project_fields.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger } from '@kbn/logging';
+import { omit } from 'lodash';
+import { BrowserSimpleFields, ConfigKey, MonitorFields } from '../../../common/runtime_types';
+import { inlineToProjectZip } from '../../common/inline_to_zip';
+
+interface MapInlineToProjectFieldsArgs {
+  monitorType: string;
+  monitor: unknown;
+  logger: Logger;
+  includeInlineScript?: boolean;
+}
+
+export async function mapInlineToProjectFields({
+  monitorType,
+  monitor,
+  logger,
+  includeInlineScript = true,
+}: MapInlineToProjectFieldsArgs) {
+  if (monitorType !== 'browser' || !monitor) return {};
+  const asBrowserMonitor = monitor as BrowserSimpleFields;
+  const inlineScript = asBrowserMonitor?.[ConfigKey.SOURCE_INLINE];
+  if (!inlineScript) return {};
+  try {
+    const projectZip = await inlineToProjectZip(
+      inlineScript,
+      asBrowserMonitor?.[ConfigKey.CONFIG_ID],
+      logger
+    );
+    if (includeInlineScript)
+      return {
+        [ConfigKey.SOURCE_INLINE]: inlineScript,
+        [ConfigKey.SOURCE_PROJECT_CONTENT]: projectZip,
+      };
+    return {
+      [ConfigKey.SOURCE_PROJECT_CONTENT]: projectZip,
+    };
+  } catch (e) {
+    logger.error(e);
+  }
+
+  return {
+    [ConfigKey.SOURCE_INLINE]: inlineScript,
+  };
+}
+
+/**
+ * We don't transmit the inline script in `source.inline.script` field anymore, because JS parsing on the backend
+ * can break on certain characters. The procedure for handling project content (the default behavior in the absence
+ * of the inline script data) is not susceptible to this issue.
+ *
+ * See https://github.com/elastic/kibana/issues/169963 for more information.
+ */
+export const dropInlineScriptForTransmission = (monitor: MonitorFields) => {
+  return (
+    monitor[ConfigKey.MONITOR_TYPE] === 'browser' &&
+    monitor[ConfigKey.MONITOR_SOURCE_TYPE] === 'ui' &&
+    !!(monitor as MonitorFields)[ConfigKey.SOURCE_INLINE] &&
+    !!(monitor as MonitorFields)[ConfigKey.SOURCE_PROJECT_CONTENT]
+      ? omit(monitor, ConfigKey.SOURCE_INLINE)
+      : monitor
+  ) as MonitorFields;
+};
```

Comment:
nit: this looks neat but a nightmare to read, i would prefer simple if else conditions :)

Reply:
won't it be enough to just check if SOURCE_PROJECT_CONTENT exists omit inline script.

Reply:
yeah, spend too much time on iterative hacking mid-PR and this is what we get 😆

Reply:
> this looks neat

I didn't even delete the brackets and `return` keyword, think how "clean" it could've been...